README.md

```diff
@@ -2,14 +2,92 @@
 
 ⚠️ **Warning: This extension is currently in alpha and subject to change before the first release.**
 
-## Installation
 
+## Installation
+Python:
+```bash
+pip install -U ndx-binned-spikes
+```
 
 ## Usage
 
+### Simple example
+The following code illustrates the use of this extension:
+
+```python
+import numpy as np
+from ndx_binned_spikes import BinnedAlignedSpikes
+
+
+data = np.array(
+    [
+        [  # Data of the first unit
+            [5, 1, 3, 2],  # First timestamp bins
+            [6, 3, 4, 3],  # Second timestamp bins
+            [4, 2, 1, 4],  # Third timestamp bins
+        ],
+        [ # Data of the second unit
+            [8, 4, 0, 2],  # First timestamp bins
+            [3, 3, 4, 2],  # Second timestamp bins
+            [2, 7, 4, 1],  # Third timestamp bins
+        ],
+    ],
+)
+
+event_timestamps = np.array([0.25, 5.0, 12.25])  # The timestamps to which we align the counts
+milliseconds_from_event_to_first_bin = -50.0  # The first bin is 50 ms before the event
+bin_width_in_milliseconds = 100.0  # Each bin is 100 ms wide
+binned_aligned_spikes = BinnedAlignedSpikes(
+    data=data,
+    event_timestamps=event_timestamps,
+    bin_width_in_milliseconds=bin_width_in_milliseconds,
+    milliseconds_from_event_to_first_bin=milliseconds_from_event_to_first_bin
+)
+
+```
+
+The resulting object is usually added to a processing module:
+
 ```python
+from pynwb.testing.mock.file import mock_NWBFile
+from pynwb import NWBHDF5IO
+
+
+nwbfile = mock_NWBFile()
 
+ecephys_processinng_module = nwbfile.create_processing_module(
+    name="ecephys", description="a description"
+)
+ecephys_processinng_module.add(binned_aligned_spikes)
 ```
 
+### Parameters and data structure
+
+In the `BinnedAlignedSpikes` class, we count events (usually spikes) around a set of timestamps that we call `event_timestamps`. That is, we align the counts to a set of event timestamps.  The process is illustrated in the diagram below.
```

Comment:
This sentence really highlights the semantics of the class name - should it be generalized to `BinnedAlignedEvents`? (as one example, the visual coding dataset has a case where sparse high activity event windows are estimated from a DF/F trace, which could have aligned representations just like this extension allows)

Also, in this paragraph, the word events is used twice in separate sentences but I think they refer to different things (one is neural events, the other is alignment events? Correct me if I'm wrong) - we should come up with a clear way of distinguishing these

Reply:
Regarding the first point. I would be fine with that.   I thought about it but I am on the edge myself as I think more general structures often are less usable and harder to discover. `ndx-binned-event-counts` and `BinnedAlignedCounts`. Maybe @bendichter has some views on this as well. 

They are referring to the same events but the current wording is confusing. I will modify it.

Reply:
OK, 
I think that I will call "the thing that you count" as spikes now to avoid the ambiguity of having two events. This will be easier to understand. I will think on a better terminology when and if we decide to generalize this to count things other than spikes.

I think it will be easier.